### ISSUE TITLE
Implement Revit => BHoM Converts for Facade_oM Objects

### ISSUE THREAD

**Issue by enarhi** [BHoM/Revit_Toolkit#890]
#### Description:
As per description, implement converts for `Facade.Elements.CurtainWall`, `Facade.Elements.Opening`, `Facade.Elements.Panel` and `Facade.Elements.FrameEdge`

**Comment by pawelbaran** [comment]
Currently curtain walls get converted to elements of type `Wall` covered with openings, each opening representing one curtain panel - big part of the job can be already done. However, I am not sure what is the relationship between `Facade.Elements.CurtainWall` vs `Physical.Elements.Wall` or `Facade.Elements.Opening` vs `Physical.Elements.Opening`. There seems to be some overlap between the disciplines.

Maybe it would be worth having a workshop on that together with @al-fisher @IsakNaslundBh @FraserGreenroyd?

**Comment by enarhi** [comment]
We've had a bit in the way of workshops already but may need a follow up as usage of Facade_oM expands. The key difference is discipline detail and requirments in the same way that a structural panel != an environmental panel. For Facade_oM, this primarily relates to additional requirements for FrameEdges, as well as the need for something like a curtain wall to contain the construction and property information for all openings and frame edges that make up said curtain wall.

**Comment by pawelbaran** [comment]
Yeah, that makes sense. Then the question is whether we can realise the above concept using the currently available disciplines, or would we need yet another one?

https://github.com/BHoM/Revit_Toolkit/blob/985422329e7df67f40172bdda6d27655c9d9c159/Revit_oM/Enums/Discipline.cs#L29-L42

**Comment by pawelbaran** [comment]
I am bumping this issue to 4.1: it will be a massive task, I am looking forward to planning it across the milestone incl. concrete tasks, people and time 👍

**Comment by enarhi** [comment]
happy with that, now that we have established more facade engine methods we will have more use case to drive revit pull development in the new year.

**Comment by enarhi** [comment]
Just noting that I am getting started on this, and as you noted @pawelbaran it will need to include the addition of Facade to the discipline Enum. Let me know if anything else needs coordinating, otherwise I will plan on updating that and digging into the needed methods to translate curtain walls, openings, and panels to facade level objects with frame edge properties and constructions accordingly.

**Comment by pawelbaran** [comment]
Good to hear you are starting with it. I am not 100% sure if the new discipline is necessary, although my intuition says you are right. Regardless of my intuition, I will play devil's advocate for a second to make sure we are not adding complexity for no valid reason: could you please name the Revit element types for which you need a new discipline, i.e. their converts could not be done within, say, `Physical`? Curtain walls and mullions should not be the case (the convert to `Facade` elements could be done within `Physical` convert), but Windows, Doors?

**Comment by enarhi** [comment]
I would think curtain walls, mullions, windows, and potentially doors. The reason for curtain walls and mullions being included is that they would contain different levels of information, including frame edge properties and profiles that are not included or needed for the current converts and which only relate to the Facade_oM elements. The physical would still return a wall, while Facade would return a Facade.CurtainWall with FrameEdgeProperties and the like.
